lua/orgmode/treesitter/listitem.lua

```diff
@@ -0,0 +1,118 @@
+local ts_utils = require('nvim-treesitter.ts_utils')
+local tree_utils = require('orgmode.utils.treesitter')
+local query = vim.treesitter.query
+local Headline = require('orgmode.treesitter.headline')
+
+local Listitem = {}
+
+function Listitem:new(listitem_node)
+  local data = { listitem = listitem_node }
+  setmetatable(data, self)
+  self.__index = self
+  return data
+end
+
+function Listitem:update_checkbox(action)
+  action = action or 'toggle'
+
+  local checkbox = self:checkbox()
+  local cookie = self:cookie()
+
+  local new_val
+  if action == 'on' then
```

Comment:
+1 on this. It will allow early returns, and maybe some of these static values can be put into a table:
```lua
local actions = {
  on = '[X]',
  off = '[ ]'
}
```

Reply:
I put it in its own function now, it's much nicer, thanks.

Currently `on` and `off` are also not used at all, I just added them because it was easy to do so, and it might be useful for some? But I could remove them as well.